src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
Ah, ok. Play will wrap it up an action asynchronously even if it is not an Action.async so there's no worry there. 

In fact, it's interesting to note that the controller function doesn't return a Result. It returns a function that accepts a Request and returns a Result and Play will schedule that function to runasynchroniusly  on its thread pool.